sdk/src/asset_handlers/font_io.rs

```diff
@@ -0,0 +1,652 @@
+    pub major_version: u16,
+    /// Minor version of the C2PA table record
+    pub minor_version: u16,
+    /// Optional URI to an active manifest
+    pub active_manifest_uri: Option<String>,
+    /// Optional embedded manifest store
+    pub manifest_store: Option<Vec<u8>>,
+}
+
+impl TableC2PA {
+    /// Creates a new C2PA table with the given values.
+    pub fn new(active_manifest_uri: Option<String>, manifest_store: Option<Vec<u8>>) -> Self {
+        Self {
+            active_manifest_uri,
+            manifest_store,
+            ..TableC2PA::default()
+        }
+    }
+
+    /// Create the checksum for this table
+    pub fn checksum(&self) -> Wrapping<u32> {
+        // // Serialize self to a throwaway stream
+        // let mut stream = Cursor::new(Vec::new());
+        // match self.write(&mut stream) {
+        //     Ok(()) => (),
+        //     Err(error) => return Err(error),
+        // }
+        // // Compute checksum of stream
+        // stream.seek(SeekFrom::Start(0)).unwrap();
+        // let mut cksum: u32 = 0;
+        // while stream.get_ref().len() > 4 {
+        //     let ckword: u32 = stream.read_u32::<BigEndian>()?;
+        //     cksum += ckword;
+        // }
+        // if stream.get_ref().len() > 0 {
+        //     let mut ckfrag: u32 = 0;
+        //     let mut factor: u32 = 256 * 256 * 256;
+        //     while stream.get_ref().len() > 0 {
+        //         let ckbyte = stream.read_u8()?;
+        //         ckfrag += ckbyte as u32 * factor;
+        //         factor /= 256;
+        //     }
+        //     cksum += ckfrag;
+        // }
+        Wrapping(0x12345678)
+    }
+
+    /// Size of this table if it were serialized right now
+    pub fn len(&self) -> usize {
+        size_of::<TableC2PARaw>()
+            + match &self.active_manifest_uri {
+                Some(uri) => uri.len(),
+                None => 0,
+            }
+            + match &self.manifest_store {
+                Some(store) => store.len(),
+                None => 0,
+            }
+    }
+
+    /// Creates a new C2PA table from the given stream.
+    pub fn make_from_reader<T: Read + Seek + ?Sized>(
+        reader: &mut T,
+        offset: u64,
+        size: usize,
+    ) -> core::result::Result<TableC2PA, Error> {
+        if size < size_of::<TableC2PARaw>() {
+            Err(Error::FontLoadError)?
+        } else {
+            let mut active_manifest_uri: Option<String> = None;
+            let mut manifest_store: Option<Vec<u8>> = None;
+            // Read the initial fixed-sized portion of the table
+            reader.seek(SeekFrom::Start(offset))?;
+            let raw_table = TableC2PARaw::new_from_reader(reader)?;
+            // Check parameters
+            if size
+                < size_of::<TableC2PARaw>()
+                    + raw_table.activeManifestUriLength as usize
+                    + raw_table.manifestStoreLength as usize
+            {
+                Err(Error::FontLoadError)?
```

Comment:
@vancejc-mt Same complaint here as you had a dozen lines back, about the needless `?` ?

Reply:
concur